vault/identity_store_oidc_provider.go

```diff
@@ -1563,6 +1579,33 @@ func (i *IdentityStore) pathOIDCAuthorize(ctx context.Context, req *logical.Requ
 		scopes:      scopes,
 	}
 
+	// Validate the optional Proof Key for Code Exchange (PKCE) if a code
+	// challenge and code challenge method are provided. See details at
+	// https://datatracker.ietf.org/doc/html/rfc7636.
+	if codeChallengeRaw, ok := d.GetOk("code_challenge"); ok {
+		codeChallenge := codeChallengeRaw.(string)
+
+		// Validate the code challenge method
+		codeChallengeMethod := d.Get("code_challenge_method").(string)
+		switch codeChallengeMethod {
+		case codeChallengeMethodPlain, codeChallengeMethodS256:
+		case "":
+			codeChallengeMethod = codeChallengeMethodPlain
```

Comment:
I would've expected `S256` here, but this default of `plain` is defined by the RFC.